### ISSUE TITLE
Grid - inline editing support

### ISSUE THREAD

**Issue by amcclain** [xh/hoist-react#883]
First step is to describe working implementations we already have in place - how they might work with / against the toolkit as currently implemented - then come up with a plan for how this could interact with support offered by inputs and forms/fields, in particular w.r.t. validation.

**Comment by lbwexler** [comment]
John and I just had a lengthy conversation about this.  We did not talk too much about the use of HoistInputs, although it does seems like it could be pretty straightforward to either build wrappers around our inputs, or modify them to effectively work as agGrid editors.

We focused on some improved "dirty/lastModified" support that we could make to our data package, so that both records and the store could have observable properties that could easily be consulted so that the GUI could easily respond to any changes made by inline editors.

The ag API is quite rich here, and it does not seem in-conflict with Grid/GridModel.  We were both leaning to continue to specify editing options directly via agOptions, rather than trying to wrap the API in some way with an additional Hoist API on column.
